src/body.rs

```diff
@@ -10,7 +13,7 @@ pub struct Body {
 
 impl Body {
     /// Instantiate a `Body` from a reader.
-    pub fn new<R: Read + 'static>(reader: R) -> Body {
+    pub fn new<R: Read + Seek + 'static>(reader: R) -> Body {
```

Comment:
I think I'd rather hold off on requiring `Seek`. There are use cases where a body could be a reader that cannot seek. Looking at golang, it seems they instead have a check if the body can be reused, and if not, just returns the `30x` response as is. I think we could probably do the same for now.